jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -102,12 +105,20 @@ public RegistryAuthenticator initialize()
    *
    * @param buildLogger the build logger used for printing messages
    * @param serverUrl the server URL for the registry (for example, {@code gcr.io})
-   * @param repository the image/repository name (also known as, namespace)
+   * @param repository the image/repository name for access (also known as, namespace)
+   * @param dependentRepositories additional image/repository names required for access; always
+   *     requested with {@code pull} scope; may be {@code null} or empty
    * @return the new {@link Initializer}
    */
   public static Initializer initializer(
-      JibLogger buildLogger, String serverUrl, String repository) {
-    return new Initializer(buildLogger, serverUrl, repository);
+      JibLogger buildLogger,
+      String serverUrl,
+      String repository,
+      @Nullable String... dependentRepositories) {
```

Comment:
Hmm.. It feels a bit off that this is both a varargs and nullable. Perhaps this should be two different `initializer` methods or it should take a `List<String>` instead (so the empty case means none)?

Reply:
I didn't like this either.  What I really want to put is a map of _repository &rarr; scope_.